src/test/java/uk/gov/justice/digital/client/dynamodb/DynamoDBClientTest.java

```diff
@@ -0,0 +1,156 @@
+        mapper = mock(ObjectMapper.class);
+        dynamoDBClient = new DynamoDBClient(dynamoDB, mapper);
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    void shouldExecuteQueryForGivenDomain() {
+        String domainTableName = "test";
+        String domainName = "incident";
+        QueryResult result = mock(QueryResult.class);
+        when(dynamoDB.query(any(QueryRequest.class))).thenReturn(result);
+        QueryResult expected_result = dynamoDBClient.executeQuery(domainTableName, domainName);
+        assert(expected_result.getItems().isEmpty());
+    }
+
+    @Test
+    void shouldReturnErrorIfDomainTableDoesntExists() {
+        String domainTableName = "test";
+        String domainName = "incident";
+        QueryResult result = mock(QueryResult.class);
+        when(dynamoDB.query(any(QueryRequest.class))).thenThrow(ResourceNotFoundException.class);
+        try {
+            QueryResult expected_result = dynamoDBClient.executeQuery(domainTableName, domainName);
+            fail();
+        } catch (AmazonDynamoDBException e) {
+            logger.info("Database not exists test");
+            logger.error(e.getMessage());
+            assertTrue(true);
+        }
+    }
+
+
+    @Test
+    @SuppressWarnings("serial")
+    void shouldReturnDomainDefinitionForGivenDomain() throws IOException {
+        DomainDefinition domainDef =  new DomainDefinition();
+        domainDef.setName("test name");
+        domainDef.setId("123");
+        domainDef.setDescription("test description");
+        QueryResult result = mock(QueryResult.class);
+        when(dynamoDBClient.executeQuery("test", "incident")).thenReturn(result);
+        List<Map<String, AttributeValue>> l = Collections.singletonList(new HashMap<String, AttributeValue>() {{
+            put("data", new AttributeValue()
+                    .withS("{\"id\": \"123\", \"name\": \"test name\", \"description\": \"test description\"}"));
+        }});
+        when(result.getItems()).thenReturn(l);
+        String data = l.get(0).get("data").getS();
+        when(mapper.readValue(data, DomainDefinition.class)).thenReturn(domainDef);
+        when(dynamoDBClient.parse(result, null)).thenReturn(domainDef);
+        DomainDefinition expectedDomainDef = dynamoDBClient.getDomainDefinition("test",
+                "incident");
+        verify(mapper, times(1)).readValue(data, DomainDefinition.class);
+        assertEquals(expectedDomainDef.getName(), "test name");
+    }
+
+
+    @Test
+    @SuppressWarnings("serial")
+    void shouldParseQueryResultForGivenDomainName() throws IOException {
+        QueryResult result = mock(QueryResult.class);
+        DomainDefinition domainDef =  new DomainDefinition();
+        domainDef.setName("test name");
+        domainDef.setId("123");
+        domainDef.setDescription("test description");
+        List<Map<String, AttributeValue>> l = Collections.singletonList(new HashMap<String, AttributeValue>() {{
```

Comment:
Use `Collections.singletonList(Collections.singletonMap("key", "value"))` instead. This will clean up the code and resolve the serialization warning.